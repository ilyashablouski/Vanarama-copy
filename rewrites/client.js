/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
