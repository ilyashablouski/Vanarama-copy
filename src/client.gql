extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
