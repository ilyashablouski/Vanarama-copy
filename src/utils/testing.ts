// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
