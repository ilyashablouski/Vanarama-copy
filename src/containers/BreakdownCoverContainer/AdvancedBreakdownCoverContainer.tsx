import React, { FC } from 'react';
import ReactMarkdown from 'react-markdown';
import Heading from '@vanarama/uibook/lib/components/atoms/heading';
import Text from '@vanarama/uibook/lib/components/atoms/text';
import Card from '@vanarama/uibook/lib/components/molecules/cards';
import Image from '@vanarama/uibook/lib/components/atoms/image';
import { getFeaturedClassPartial } from '../../utils/layout';
import { advancedBreakdownCoverPage_advancedBreakdownCoverPage_sections as Section } from '../../../generated/advancedBreakdownCoverPage';

interface IProps {
  sections: Section | null;
  title: string | null;
  body: string | null;
}

const AdvancedBreakdownCoverContainer: FC<IProps> = ({
  title,
  body,
  sections,
}) => {
  const featured1 = sections?.featured1;
  const tiles = sections?.tiles;
  const featured2 = sections?.featured2;
  const featured3 = sections?.featured3;

  const featured1Html = (
    <>
      {featured1 && (
        <section className={`row:${getFeaturedClassPartial(featured1)}`}>
          <div>
            <Heading
              color="black"
              size="lead"
              tag={featured1.titleTag as keyof JSX.IntrinsicElements}
            >
              {featured1.title}
            </Heading>
            <Text color="darker" size="regular">
              <ReactMarkdown
                source={featured1.body?.replace(/\n/gi, '&nbsp;\n') || ''}
              />
            </Text>
          </div>
          <div>
            {featured1.image?.file?.url && (
              <Image
                src={featured1.image?.file?.url}
                alt={featured1.image?.file?.fileName}
              />
            )}
          </div>
        </section>
      )}
    </>
  );

  const tilesHtml = (
    <>
      {tiles && (
        <section className="row:bg-light">
          <div>
            <Heading
              color="black"
              size="large"
              tag={tiles.titleTag as keyof JSX.IntrinsicElements}
            >
              {tiles.tilesTitle}
            </Heading>
            {tiles?.tiles?.length && (
              <div className="row:cards-2col">
                {tiles?.tiles.map((el, idx) => (
                  <Card
                    inline
                    key={el.title || idx}
                    imageSrc={el.image?.file?.url || ''}
                    title={{
                      title: el.title || '',
                      description: el.body || '',
                    }}
                    className="breakdown"
                  />
                ))}
              </div>
            )}
          </div>
        </section>
      )}
    </>
  );

  const featured2Html = (
    <>
      {featured2 && (
        <section className={`row:${getFeaturedClassPartial(featured2)}`}>
          <div>
            <Heading
              color="black"
              size="large"
              tag={featured2.titleTag as keyof JSX.IntrinsicElements}
            >
              {featured2.title}
            </Heading>
            <Text color="darker" size="regular">
              <ReactMarkdown
                source={featured2.body?.replace(/\n/gi, '&nbsp;\n') || ''}
              />
            </Text>
          </div>
          <div>
            {featured2.image?.file?.url && (
              <Image
                src={featured2.image?.file?.url}
                alt={featured2.image?.file?.fileName}
              />
            )}
          </div>
        </section>
      )}
    </>
  );

  const featured3Html = (
    <>
      {featured3 && (
        <section className={`row:${getFeaturedClassPartial(featured3)}`}>
          <div>
            <Heading
              color="black"
              size="large"
              tag={featured3.titleTag as keyof JSX.IntrinsicElements}
            >
              {featured3.title}
            </Heading>
            <Text color="darker" size="regular">
              <ReactMarkdown
                source={featured3.body?.replace(/\n/gi, '&nbsp;\n') || ''}
              />
            </Text>
          </div>
          <div>
            {featured3.image?.file?.url && (
              <Image
                src={featured3.image?.file?.url}
                alt={featured3.image?.file?.fileName}
              />
            )}
          </div>
        </section>
      )}
    </>
  );

  return (
    <>
      <div className="row:title">
        <Heading size="xlarge" color="black" tag="h1">
          {title}
        </Heading>
        <ReactMarkdown source={body || ''} />
      </div>
      {featured1Html}
      {tilesHtml}
      {featured2Html}
      {featured3Html}
    </>
  );
};

export default AdvancedBreakdownCoverContainer;
