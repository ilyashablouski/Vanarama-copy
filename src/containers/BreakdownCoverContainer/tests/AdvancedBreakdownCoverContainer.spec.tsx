import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { screen, render, waitFor } from '@testing-library/react';
import AdvancedBreakdownCoverContainer from '../AdvancedBreakdownCoverContainer';

// ARRANGE
const TITLE = `Advanced Breakdown Cover`;
const BODY = ``;

const SECTIONS = {
  __typename: 'Sections',
  featured1: {
    __typename: 'Featured',
    title: 'Advanced Breakdown Cover',
    titleTag: 'h2',
    body:
      "## Full & Drama-Free Breakdown Cover For Less\n\nBreaking down is the last thing you want to think about when you order your brand-new van or pickup truck from us. But it could happen, & if you don't have breakdown cover, you're likely to find yourself forking out a fortune to recover your vehicle & find a replacement.\n\nSo, get our Advanced Breakdown Cover when you lease a brand-new van or pickup truck from Vanarama. If you break down, you're covered! If your vehicle needs repairing we'll give you a similar-sized vehicle for up to 7 days to keep your business moving!\n\n[View Deals](https://beta.vanarama.com/special-offers.html)",
    image: {
      __typename: 'Image',
      title: 'abc-man',
      description: null,
      file: {
        __typename: 'File',
        url:
          '//images.ctfassets.net/3xid768u5joa/5Z6WeTc9kvkrnJpomCE1s4/0a3ec860b45748e384761e24be3d2e42/abc-man.jpg',
        fileName: 'abc-man.jpg',
        contentType: 'image/jpeg',
      },
    },
  },
  featured2: {
    __typename: 'Featured',
    title: 'How Much Will It Cost You?',
    titleTag: 'h2',
    body:
      'Advanced Breakdown Cover is only £9 per month for the entire length of your lease. You get all that piece of mind for less than a tenner a month... talk about a no-brainer.\n\n[View Deals](https://beta.vanarama.com/special-offers.html)',
  },
  tiles: {
    __typename: 'Tiles',
    name: "What's Included?",
    tilesTitle: "What's Included?",
    titleTag: 'h2',
    tiles: [
      {
        __typename: 'Tile',
        title: 'Breakdown Cover',
        body:
          "Wherever your vehicle breaks down, our trusted partner, the RAC, will provide assistance, so you'll never be left stranded.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Fleet-Icon',
          description: 'Fleet-Icon',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/3a609lEJL2WBF9fBG9V6N6/9bbcee8eba6f7deecfbbab92582283e2/Fleet-Icon.png',
            fileName: 'Fleet-Icon.png',
            contentType: 'image/png',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'At Home Cover',
        body:
          "If your vehicle won't start, the RAC will fix your vehicle if you break down at, or within ¼ mile of home. Just pop the kettle on & sit tight.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'free-safe-contactless-delivery',
          description: 'free-safe-contactless-delivery',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/7gH6pHpHcgXHt14wJlfqmz/0acc5f3d654e0a2f1e2e1e584901983e/delivery.jpg',
            fileName: 'delivery.jpg',
            contentType: 'image/jpeg',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'Recovery',
        body:
          "If your vehicle can't be fixed at the roadside, the RAC will take it for repair. All part of the service!",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Insurance-Icon',
          description: 'Insurance-Icon',
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/3KhuHOZWIeppLEsCTcouiY/a384211adc3f246728f3c5fd9d8cf637/Insurance-Icon.png',
            fileName: 'Insurance-Icon.png',
            contentType: 'image/png',
          },
        },
      },
      {
        __typename: 'Tile',
        title: 'Replacement Vehicle',
        body:
          "If your vehicle's off the road needing repair, we'll keep your business moving by giving you a similar-sized vehicle for up to 7 days.",
        link: null,
        image: {
          __typename: 'Image',
          title: 'Vanarama-Price-Promise-Icon',
          description: null,
          file: {
            __typename: 'File',
            url:
              '//images.ctfassets.net/3xid768u5joa/4v2ALClaL6YnE0NYRsd8uc/7e92f2b83eb9e594ced8269152961026/Icon-PricePromise-RGB.png',
            fileName: 'Icon-PricePromise-RGB.png',
            contentType: 'image/png',
          },
        },
      },
    ],
  },
} as any;

describe('<AdvancedBreakdownCoverContainer />', () => {
  it('should match snapshot', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider addTypename={false}>
        <AdvancedBreakdownCoverContainer
          sections={SECTIONS}
          title={TITLE}
          body={BODY}
        />
      </MockedProvider>,
    );
    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(`Full & Drama-Free Breakdown Cover For Less`),
      ).toBeInTheDocument();
    });
    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
