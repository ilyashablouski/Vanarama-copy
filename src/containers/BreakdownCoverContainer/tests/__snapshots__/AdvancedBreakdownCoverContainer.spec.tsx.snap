// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedBreakdownCoverContainer /> should match snapshot 1`] = `
<body>
  <div>
    <div
      class="row:title"
    >
      <h1
        class="heading -black -xlarge"
      >
        Advanced Breakdown Cover
      </h1>
    </div>
    <section
      class="row:featured-right"
    >
      <div>
        <h2
          class="heading -black -large"
        >
          Advanced Breakdown Cover
        </h2>
        <span
          class="text -regular -darker"
        >
          <h2>
            Full & Drama-Free Breakdown Cover For Less
             
          </h2>
          <p>
             
            
Breaking down is the last thing you want to think about when you order your brand-new van or pickup truck from us. But it could happen, & if you don't have breakdown cover, you're likely to find yourself forking out a fortune to recover your vehicle & find a replacement.
             
            

             
            
So, get our Advanced Breakdown Cover when you lease a brand-new van or pickup truck from Vanarama. If you break down, you're covered! If your vehicle needs repairing we'll give you a similar-sized vehicle for up to 7 days to keep your business moving!
             
            

             
            

            <a
              href="https://beta.vanarama.com/special-offers.html"
            >
              View Deals
            </a>
          </p>
        </span>
      </div>
      <div>
        <div
          class="image -expand"
        >
          <img
            alt="abc-man.jpg"
            class="image--native"
            src="//images.ctfassets.net/3xid768u5joa/5Z6WeTc9kvkrnJpomCE1s4/0a3ec860b45748e384761e24be3d2e42/abc-man.jpg"
          />
        </div>
      </div>
    </section>
    <section
      class="row:bg-light"
    >
      <div>
        <h2
          class="heading -black -large"
        >
          What's Included?
        </h2>
        <div
          class="row:cards-2col"
        >
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              alt="img"
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/3a609lEJL2WBF9fBG9V6N6/9bbcee8eba6f7deecfbbab92582283e2/Fleet-Icon.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Breakdown Cover
              </span>
              <span
                class="heading -dark -small"
                data-testid="card-description"
              >
                Wherever your vehicle breaks down, our trusted partner, the RAC, will provide assistance, so you'll never be left stranded.
              </span>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              alt="img"
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/7gH6pHpHcgXHt14wJlfqmz/0acc5f3d654e0a2f1e2e1e584901983e/delivery.jpg"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                At Home Cover
              </span>
              <span
                class="heading -dark -small"
                data-testid="card-description"
              >
                If your vehicle won't start, the RAC will fix your vehicle if you break down at, or within ¼ mile of home. Just pop the kettle on & sit tight.
              </span>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              alt="img"
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/3KhuHOZWIeppLEsCTcouiY/a384211adc3f246728f3c5fd9d8cf637/Insurance-Icon.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Recovery
              </span>
              <span
                class="heading -dark -small"
                data-testid="card-description"
              >
                If your vehicle can't be fixed at the roadside, the RAC will take it for repair. All part of the service!
              </span>
            </div>
          </div>
          <div
            class="card breakdown -inline"
            data-testid="card"
          >
            <img
              alt="img"
              class="card-image"
              data-testid="card-image"
              src="//images.ctfassets.net/3xid768u5joa/4v2ALClaL6YnE0NYRsd8uc/7e92f2b83eb9e594ced8269152961026/Icon-PricePromise-RGB.png"
            />
            <div
              class="title"
            >
              <span
                class="heading -black -lead"
                data-testid="card-heading"
              >
                Replacement Vehicle
              </span>
              <span
                class="heading -dark -small"
                data-testid="card-description"
              >
                If your vehicle's off the road needing repair, we'll keep your business moving by giving you a similar-sized vehicle for up to 7 days.
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="row:featured-right"
    >
      <div>
        <h2
          class="heading -black -large"
        >
          How Much Will It Cost You?
        </h2>
        <span
          class="text -regular -darker"
        >
          <p>
            Advanced Breakdown Cover is only £9 per month for the entire length of your lease. You get all that piece of mind for less than a tenner a month... talk about a no-brainer.
             
            

             
            

            <a
              href="https://beta.vanarama.com/special-offers.html"
            >
              View Deals
            </a>
          </p>
        </span>
      </div>
      <div />
    </section>
  </div>
</body>
`;
